master/buildbot/test/fake/gce.py

```diff
@@ -0,0 +1,169 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from buildbot.util import gceclientservice
+
+
+class GCEAsyncResult:
+    def __init__(self, record_id):
+        self.record_id = record_id
+
+
+class ExpectedRequest:
+    NORMAL = 1
+    UNTIL = 2
+
+    def __init__(self, mode, httpMethod, url, params={}, json={}, result={}, data=None, resultCode=200):
+        self.mode = mode
+        self.httpMethod = httpMethod
+        self.url = url
+        self.params = params
+        self.json = json
+        self.data = data
+        self.result = result
+        self.resultCode = resultCode
+
+
+class GCERecorder(gceclientservice.GCEClientService):
```

Comment:
I'd prefer something like `FakeGCEClientService`. Right now the naming is such that it seems the class implements functionality related to GCE when in fact it's just some test hooks.